samples/javascript_nodejs/08.suggested-actions/bot.js

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityTypes, MessageFactory, TurnContext } = require('botbuilder');
+
+/**
+ * A bot that responds to input from suggested actions.
+ */
+class SuggestedActionsBot {
+    /**
+     * Every conversation turn for our SuggestedActionsbot will call this method.
+     * There are no dialogs used, since it's "single turn" processing, meaning a single request and
+     * response, with no stateful conversation.
+     * @param {TurnContext} turnContext A TurnContext instance containing all the data needed for processing this conversation turn.
+     */
+    async onTurn(turnContext) {
+        // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
+        if (turnContext.activity.type === ActivityTypes.Message) {
+            const text = turnContext.activity.text;
+
+            // Create an array with the valid color options.
+            const validColors = ['Red', 'Blue', 'Yellow'];
+
+            // If the `text` is in the Array, a valid color was selected and send agreement.
+            if (validColors.includes(text)) {
+                await turnContext.sendActivity(`I agree, ${ text } is the best color.`);
+            } else {
+                await turnContext.sendActivity('Please select a color.');
+            }
+
+            // After the bot has responded send the suggested actions.
+            await this.sendSuggestedActions(turnContext);
+        } else if (turnContext.activity.type === ActivityTypes.ConversationUpdate) {
+            await this.sendWelcomeMessage(turnContext);
+        } else {
+            await turnContext.sendActivity(`[${ turnContext.activity.type } event detected.]`);
+        }
+    }
+
+    /**
+     * Send a welcome message along with suggested actions for the user to click.
+     * @param {TurnContext} turnContext A TurnContext instance containing all the data needed for processing this conversation turn.
+     */
+    async sendWelcomeMessage(turnContext) {
```

Comment:
Can we simplify this? 

if (turnContext.activity.membersAdded) {
            // Iterate over all new members added to the conversation
            for (var idx in turnContext.activity.membersAdded) {
                if (turnContext.activity.membersAdded[idx].id !== turnContext.activity.recipient.id) {
                    await turnContext.sendActivity(`foo`);
                }
            }
        }

Reply:
Done!